Dockerfile

```diff
@@ -3,20 +3,22 @@
 #   The packages are installed in the `/autoinstrumentation` directory. This is required as when instrumenting the pod by CWOperator,
 #   one init container will be created to copy all the content in `/autoinstrumentation` directory to app's container. Then
 #   update the `PYTHONPATH` environment variable accordingly. Then in the second stage, copy the directory to `/autoinstrumentation`.
-
-# Using Python 3.10 because we are utilizing the opentelemetry-exporter-otlp-proto-grpc exporter,
-# which relies on grpcio as a dependency. grpcio has strict dependencies on the OS and Python version.
-# Also mentioned in Docker build template in the upstream repository:
-# https://github.com/open-telemetry/opentelemetry-operator/blob/b5bb0ae34720d4be2d229dafecb87b61b37699b0/autoinstrumentation/python/requirements.txt#L2
-# For further details, please refer to: https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/recover-python-functions.md#the-python-interpre[…]tions-python-worker
-FROM python:3.10 AS build
+FROM python:3.11 AS build
 
 WORKDIR /operator-build
 
 ADD aws-opentelemetry-distro/ ./aws-opentelemetry-distro/
 
 RUN mkdir workspace && pip install --target workspace ./aws-opentelemetry-distro
 
+# Remove opentelemetry-exporter-otlp-proto-grpc and grpcio, as grpcio has strict dependencies on the Python version and
+# will cause confusing failures if gRPC protocol is used. Now if gRPC protocol is requested by the user, instrumentation
+# will complain that grpc is not installed, which is more understandable. References:
+# * https://github.com/open-telemetry/opentelemetry-operator/blob/b5bb0ae34720d4be2d229dafecb87b61b37699b0/autoinstrumentation/python/requirements.txt#L2
+# * https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/azure-functions/recover-python-functions.md#troubleshoot-cannot-import-cygrpc
+RUN pip uninstall opentelemetry-exporter-otlp-proto-grpc -y
+RUN pip uninstall grpcio -y
```

Comment:
The `opentelemetry-exporter-otlp-proto-grpc` installs other dependencies as well. Do you think we should remove those as well? Or just the problematic `grpcio` for now?
IMO, just removing the `grpcio` should be fine but want to double-check.
```
(venv) ➜  aws-otel-python-instrumentation git:(main) ✗ pipdeptree                                                
opentelemetry-exporter-otlp-proto-grpc==1.22.0
├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
├── googleapis-common-protos [required: ~=1.52, installed: 1.63.0]
│   └── protobuf [required: >=3.19.5,<5.0.0.dev0,!=4.21.5,!=4.21.4,!=4.21.3,!=4.21.2,!=4.21.1,!=3.20.1,!=3.20.0, installed: 4.25.3]
├── grpcio [required: >=1.0.0,<2.0.0, installed: 1.62.2]
├── opentelemetry-api [required: ~=1.15, installed: 1.22.0]
│   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
│   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
│   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
│       └── zipp [required: >=0.5, installed: 3.18.1]
├── opentelemetry-exporter-otlp-proto-common [required: ==1.22.0, installed: 1.22.0]
│   ├── backoff [required: >=1.10.0,<3.0.0, installed: 2.2.1]
│   └── opentelemetry-proto [required: ==1.22.0, installed: 1.22.0]
│       └── protobuf [required: >=3.19,<5.0, installed: 4.25.3]
├── opentelemetry-proto [required: ==1.22.0, installed: 1.22.0]
│   └── protobuf [required: >=3.19,<5.0, installed: 4.25.3]
└── opentelemetry-sdk [required: ~=1.22.0, installed: 1.22.0]
    ├── opentelemetry-api [required: ==1.22.0, installed: 1.22.0]
    │   ├── Deprecated [required: >=1.2.6, installed: 1.2.14]
    │   │   └── wrapt [required: >=1.10,<2, installed: 1.16.0]
    │   └── importlib-metadata [required: >=6.0,<7.0, installed: 6.11.0]
    │       └── zipp [required: >=0.5, installed: 3.18.1]
    ├── opentelemetry-semantic-conventions [required: ==0.43b0, installed: 0.43b0]
    └── typing_extensions [required: >=3.7.4, installed: 4.11.0]

```

Reply:
I think all of those are shared with `opentelemetry-exporter-otlp-proto-http`:
```
> pip show opentelemetry-exporter-otlp-proto-http
Name: opentelemetry-exporter-otlp-proto-http
Version: 1.22.0
Summary: OpenTelemetry Collector Protobuf over HTTP Exporter
Home-page: 
Author: 
Author-email: OpenTelemetry Authors <cncf-opentelemetry-contributors@lists.cncf.io>
License: 
Location: /local/home/thp/.local/lib/python3.9/site-packages
Requires: backoff, deprecated, googleapis-common-protos, opentelemetry-api, opentelemetry-exporter-otlp-proto-common, opentelemetry-proto, opentelemetry-sdk, requests
Required-by: aws-opentelemetry-distro

> pip show opentelemetry-exporter-otlp-proto-grpc
Name: opentelemetry-exporter-otlp-proto-grpc
Version: 1.22.0
Summary: OpenTelemetry Collector Protobuf over gRPC Exporter
Home-page: 
Author: 
Author-email: OpenTelemetry Authors <cncf-opentelemetry-contributors@lists.cncf.io>
License: 
Location: /local/home/thp/.local/lib/python3.9/site-packages
Requires: backoff, deprecated, googleapis-common-protos, grpcio, opentelemetry-api, opentelemetry-exporter-otlp-proto-common, opentelemetry-proto, opentelemetry-sdk
Required-by: aws-opentelemetry-distro
```

No action required.